src/network/networkd-dhcp4.c

```diff
@@ -583,6 +586,42 @@ static int dhcp4_set_hostname(Link *link) {
         return sd_dhcp_client_set_hostname(link->dhcp_client, hn);
 }
 
+static bool check_promote_secondaries(const char *ifname ) {
+        _cleanup_free_ char *promote_secondaries_sysctl = NULL,*promote_secondaries_path = NULL;
+
+        promote_secondaries_path = strjoin("net/ipv4/conf/", ifname, "/promote_secondaries");
```

Comment:
given that this is bounded, maybe use "strjoina()" rather than "strjoin()". i.e. use stack rather than heap allocation

Reply:
Thanks, done here and below.